pintc/src/predicate/optimize/const_folding.rs

```diff
@@ -70,6 +82,7 @@ pub(crate) fn fold_consts(contract: &mut Contract) -> bool {
             did_fold_const = true;
         }
         expr_keys_to_replace = FxHashMap::default();
+        scope_values = FxHashMap::default();
```

Comment:
So might as well declare `scope_values` inside the `for` loop to avoid paying the cost of clearing it.